### PR TITLE
Update release to build darwin/arm64 binary

### PR DESCRIPTION
**Pull request by briandealwis**
Towards #5042 

We currently build with cgo to enable the fast [notify](https://github.com/rjeczalik/notify) trigger.  But [`dockercore/golang-cross`](https://hub.docker.com/r/dockercore/golang-cross) has too old a toolchain to enable cgo for darwin/arm64.  So for darwin/arm64 we disable cgo and use notify's kqueue support instead.

This is a stop-gap measure since we need to produce signed binaries anyways (#2154).

This PR:
- Tweaks the Makefile variables to allow specifying the Go version, build tags, and CGO_ENABLED on a per os/arch basis.  Reorganize the variables on a per os/arch basis.
- Updates github.com/rjeczalik/notify to track master to avoid explicitly specifying `kqueue` (https://github.com/rjeczalik/notify/pull/182) and also to pull in bug fixes that are not yet in a release.
- Uses github.com/randall77/makefat to build a multi-arch ("fat") binary for Darwin for those that want it.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,31 +42,31 @@ ifeq "$(strip $(VERSION))" ""
...
+GO_BUILD_TAGS_darwin = release
 
 ifneq "$(strip $(LOCAL))" "true"
 	override STATIK_FILES =  cmd/skaffold/app/cmd/statik/statik.go
 endif
 
 # when build for local development (`LOCAL=true make install` can skip license check)
 $(BUILD_DIR)/$(PROJECT): $(STATIK_FILES) $(GO_FILES) $(BUILD_DIR)
-	GOOS=$(GOOS) GOARCH=$(GOARCH) CGO_ENABLED=1 go build -gcflags="all=-N -l" -tags $(GO_BUILD_TAGS_$(GOOS)) -ldflags $(GO_LDFLAGS_$(GOOS)) -o $@ $(BUILD_PACKAGE)
+	GOOS=$(GOOS) GOARCH=$(GOARCH) CGO_ENABLED=1 go build -gcflags="all=-N -l" -tags "$(GO_BUILD_TAGS_$(GOOS))" -ldflags "$(GO_LDFLAGS_$(GOOS))" -o $@ $(BUILD_PACKAGE)
```
Comment (IsaacPD) [inline Makefile:69]: ```GO_LDFLAGS_$(GOOS)``` doesn't seem to be defined for windows, darwin, or linux after these changes, are they supposed to be defined somewhere here?

Comment (briandealwis) [reply Makefile:69]: Hmm you're right — I've missed something here.

**Comment by briandealwis** [comment]
Converting this to a draft as I'm not entirely sure if it's worth undertaking a stop-gap measure vs just doing the right thing in the first place.

**Comment by briandealwis** [comment]
Converting this from draft: although using a CGO-less build is a stop-gap measure, it gets a native binary out.  A perfect solution can be brought in with #2154.

**Review by IsaacPD — APPROVED** [review]
I was able to build the darwin/arm64 binary, don't have a machine to test the actual binary on though, LGTM!

**Pull request merged**